### PR TITLE
Disallow initializers for extern events

### LINKED ISSUE #40092: Extern events should not be allowed to have initializers
**Version Used**: 
3.5.0-beta1-19568-03 (b4c245ac)

**Steps to Reproduce**:

```cs
using System;
public class Class1 {
    extern event Action Event1 = null;
}
```

**Expected Behavior**:

Should have an error similar to `error CS0074: 'Class1.Event1': abstract event cannot have initializer`, but referring to 'extern event' instead of 'abstract event'

**Actual Behavior**:
Produces an assembly which decompiles approximately to the following:

```cs
public class Class1
{
    [CompilerGenerated]
    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
    private Action m_Event1 = null;

    private event Action Event1
    {
        [CompilerGenerated]
        add;
        [CompilerGenerated]
        remove;
    }
}
```

**Spec language**:

https://github.com/dotnet/csharplang/blob/master/spec/classes.md#events

> It is a compile-time error for a *variable_declarator* of an event declaration with an `abstract` or `external` modifier to include a *variable_initializer*.

### PR DESCRIPTION
**Pull request by RikkiGibson**
Closes #40092

### CODE REVIEW

**Comment by AlekseyTs** [comment]
This feels like a breaking change. Has it been approved by a Compat Council? Should also add it to the list of breaking changes.

File: src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs [hunk 1]
```diff
@@ -1058,6 +1058,26 @@ abstract class Test
...
+    public extern event D e = null; // 1, 2
+}
+";
+            CreateCompilation(text).VerifyDiagnostics(
+                // (6,27): error CS8800: 'Test.e': extern event cannot have initializer
+                //     public extern event D e = null; // 1, 2
+                Diagnostic(ErrorCode.ERR_ExternEventInitializer, "e").WithArguments("Test.e").WithLocation(6, 27),
+                // (6,27): warning CS0414: The field 'Test.e' is assigned but its value is never used
+                //     public extern event D e = null; // 1, 2
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "e").WithArguments("Test.e").WithLocation(6, 27));
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs]: >WRN_UnreferencedFieldAssg [](start = 37, length = 25)

Consider adding a pragma to suppress this warning

File: src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs [hunk 2]
```diff
@@ -1058,6 +1058,26 @@ abstract class Test
                 Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "e").WithArguments("Test.e").WithLocation(6, 29));
         }
 
+        [Fact]
+        public void ExternEventInitializer()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs]: >ExternEventInitializer [](start = 20, length = 22)

Consider moving this test to src\Compilers\CSharp\Test\Symbol\Symbols\Source\EventTests.cs

File: src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs [hunk 3]
```diff
@@ -1058,6 +1058,26 @@ abstract class Test
                 Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "e").WithArguments("Test.e").WithLocation(6, 29));
         }
 
+        [Fact]
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs]: >[Fact] [](start = 8, length = 6)

Consider adding WorkItem

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

**Comment by RikkiGibson** [comment]
It seems like we have consensus from the Compat Council that we should take this breaking change.

**Pull request merged**